tests/specs/mcd/verification.k

```diff
@@ -309,4 +309,17 @@ module VERIFICATION
     rule X -Word Y <=Int X => #rangeUInt(256, X -Int Y) requires #rangeUInt(256, X) andBool #rangeUInt(256, Y)
     rule 0 <=Int X +Int Y => true requires 0 <=Int X andBool 0 <=Int Y
 
+    // ### flipper-addu48u48-fail-rough
+
+    rule maxUInt48 &Int (X +Int Y) <Int X => true requires #rangeUInt(48, X) andBool #rangeUInt(48, Y) andBool notBool (X +Int Y <Int pow48)
+
+    // ### vat-subui-fail-rough
+
+    rule 0 s<Word X => bool2Word(sgn(X) ==Int 1 andBool X =/=Int 0) requires #rangeUInt(256, X)
+    rule sgn(chop(Y)) ==K 1 => 0 <=Int Y requires #rangeSInt(256, Y)
+
+    // ### vat-subui-pass-rough
+
+    rule X -Word chop(Y) => chop(X -Int Y)
+
```

Comment:
z3 proofs for the last three lemmas: https://rise4fun.com/Z3/LFIz

Reply:
As I suggested before, please try to write a z3 proof for lemmas involving the modular arithmetic.

Reply:
I've included your proofs in this PR. I figured these ones were simple enough it wasn't necessary. I do like the idea of each lemma being attached to a Z3 query that has to be discharged for the lemma to hold.